avaframe/ana3AIMEC/aimecTools.py

```diff
@@ -0,0 +1,933 @@
+    rasterArea = rasterTransfo['rasterArea']
+
+    # initialize Arrays
+    nTopo = len(transformedRasters)
+    maxaCrossMax = np.empty((nTopo))
+    aCrossMax = np.zeros((nTopo, len(scoord)))
+    aCrossMean = np.zeros((nTopo, len(scoord)))
+    log.debug('Analyzing %s' % (dataType))
+    log.debug('{: <10} {: <10}'.format('Sim number ', 'maxCrossMax '))
+    # For each data set
+    for i in range(nTopo):
+        rasterData = transformedRasters[i]
+
+        # Max Mean in each Cross-Section for each field
+        maxaCrossMax[i], aCrossMax[i], aCrossMean[i] = getMaxMeanValues(rasterData, rasterArea)
+        log.debug('{: <10} {:<10.4f}'.format(*[i+1, maxaCrossMax[i]]))
+
+    return maxaCrossMax, aCrossMax, aCrossMean
+
+
+def analyzeArea(rasterTransfo, runoutLength, pLim, dataPressure, cfgPath, cfgFlags):
+    """Compare results to reference.
+
+    Compute True positive, False negative... areas.
+
+    Parameters
+    ----------
+    rasterTransfo: dict
+        transformation information
+    resAnalysis: dict
+        resAnalysis dictionary containing all results to update
+    pLim: float
+        numerical value of the pressure limit to use
+    dataPressure: list
+        list of transformed pressure rasters
+    cfgPath: dict
+        path to data to analyse
+    cfgFlags: configparser
+        configparser with plot and write flags
+
+    Returns
+    -------
+    TP: float
+        ref = True sim2 = True
+    FN: float
+        ref = False sim2 = True
+    FP: float
+        ref = True sim2 = False
+    TN: float
+        ref = False sim2 = False
+    """
+    nRef = cfgPath['referenceFile']
+    cellarea = rasterTransfo['rasterArea']
+    indStartOfRunout = rasterTransfo['indStartOfRunout']
+
+    # initialize Arrays
+    nTopo = len(dataPressure)
+    TP = np.empty((nTopo))
+    FN = np.empty((nTopo))
+    FP = np.empty((nTopo))
+    TN = np.empty((nTopo))
+
+    # take first simulation as reference
+    refMask = copy.deepcopy(dataPressure[nRef])
+    # prepare mask for area resAnalysis
+    refMask = np.where(np.isnan(refMask), 0, refMask)
+    refMask = np.where(refMask < pLim, 0, refMask)
+    refMask = np.where(refMask >= pLim, 1, refMask)
+    # comparison rasterdata with mask
+    log.debug('{: <15} {: <15} {: <15} {: <15} {: <15}'.format(
+        'Sim number ', 'TP ', 'FN ', 'FP ', 'TN'))
+    # rasterinfo
+    nStart = indStartOfRunout
+    # inputs for plot
+    inputs = {}
+    inputs['runoutLength'] = runoutLength
+    inputs['pressureLimit'] = pLim
+    inputs['refDataPressure'] = dataPressure[nRef]
+    inputs['refRasterMask'] = refMask
+    inputs['nStart'] = nStart
+
+    for i in range(nTopo):
+        rasterdata = dataPressure[i]
+
+        """
+        area
+        # true positive: reality(mask)=1, model(rasterdata)=1
+        # false negative: reality(mask)=1, model(rasterdata)=0
+        # false positive: reality(mask)=0, model(rasterdata)=1
+        # true negative: reality(mask)=0, model(rasterdata)=0
```

Comment:
Mh, maybe change reality to result1 and model to result2 ?

Reply:
:+1:

Reply:
done